### PR TITLE
BOCM-276

### PR DESCRIPTION
**Pull request by brpins**
## Issue ticket number and link

https://pins-ds.atlassian.net/browse/BOCM-276

## Type of change 🧩

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update
- [ ] Other (please explain in the description section above)

## Checklist before requesting a review

- [x] I have performed a self-review of my own code
- [x] I have double checked this work does not include any hardcoded secrets or passwords
- [x] I have made corresponding changes to the documentation
- [x] I have provided details on how I have tested my code
- [x] I have referenced the ticket number above
- [ ] I have provided a description of the ticket
- [x] I have included unit tests to cover any testable code changes

### CODE REVIEW

File: apps/api/src/server/app/inspector/inspector.validators.js [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+export const validateUserId = async (request, response, next) => {
+	const result = await header('userId')
+		.notEmpty()
+		.bail()
+		.withMessage('Authentication error. Missing header `userId`.')
+		.toInt()
+		.run(request);
+
+	if (!result.isEmpty()) {
+		response.status(401).send({ errors: result.formatWith(({ msg }) => msg).mapped() });
```
Comment (mayya-bond-kc) [inline apps/api/src/server/app/inspector/inspector.validators.js:20]: Not sure what your approach to error messaging is. I would have assumed that if we have an error handler, then all errors should go through it for consistency. Therefore, here I would throw an error

File: apps/api/src/server/app/inspector/inspector.validators.js [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+ */
+function handleValidationError (request, response, next) {
+	const result = validationResult(request).formatWith(({ msg }) => msg);
+
+	if (!result.isEmpty()) {
+		response.status(400).send({ errors: result.mapped() });
+	} else {
+		next();
+	}
+}
```
[the remark this answers is not in the document]
Comment (brpins) [reply apps/api/src/server/app/inspector/inspector.validators.js:94]: Yeah. Same with `validateUserBelongsToAppeal` too probably.

File: apps/api/src/server/app/inspector/inspector.validators.js (hunk 1 above)
Comment (brpins) [reply apps/api/src/server/app/inspector/inspector.validators.js:20]: yeah, we can do that. So you want to:

`throw new InspectorError({ errors: result.formatWith(({ msg }) => msg).mapped() }, 401);` 

right?

File: apps/api/src/server/app/inspector/inspector.validators.js [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+	validateUserId,
+	async (request, response, next) => {
+		const result = await header('userId')
+			.custom(async (/** @type {number} */ userId, { req }) => {
+				const appeal = await appealRepository.getById(req.params.appealId);
+
+				if (appeal.userId === userId) {
+					return true;
+				}
+				throw new Error('User is not permitted to perform this action.');
```
[the remark this answers is not in the document]
Comment (brpins) [reply apps/api/src/server/app/inspector/inspector.validators.js:36]: I think it depends. In this case, this error will only be collected by express-validator to trigger a failed validation message, so a generic `Error` makes sense for this purpose.

**Pull request merged**